### PR TITLE
Implement `addon:config` and `addon:ls`

### PR DESCRIPTION
**Pull request by zzxwill**
Store addon center configuration in ~/.rudr/addon_config;
list workloaddefinitions and traitdefinitions, and check
wheter they are intalled.

### CODE REVIEW

**Comment by zzxwill** [comment]
Here are all tests cases.
```
$ rudr addon:config
Please set addon center, `local` or an URL.%

$ rudr addon:config "https://api.github.com/repos/zzxwill/catalog/contents/repository?ref=plugin"
Successfully configured Addon center: https://api.github.com/repos/zzxwill/catalog/contents/repository?ref=plugin

$ rudr addon:config "https://api.github.com/repos/zzxwill/catalog/contents/repository?ref=plugin" def
Unnecessary arguments are specified, please try again%


$ rudr addon:ls
NAME        	TYPE    	DEFINITION                      	STATUS     	APPLIES-TO
ManualScaler	trait   	manualscalertraits.core.oam.dev 	installed  	core.oam.dev/v1alpha2.ContainerizedWorkload
expose      	trait   	serviceexposes.core.oam.dev     	uninstalled
cloneset    	workload	clonesetworkloads.apps.kruise.io	uninstalled	-
statefulset 	workload	statefulsets.apps               	uninstalled	-
```

**Comment by zzxwill** [comment]
I need to refine the code based on CUE implementation.

**Comment by zzxwill** [comment]
The addon center repository is https://api.github.com/repos/zzxwill/catalog/contents/repository?ref=plugin and the manifests are located at https://github.com/zzxwill/catalog/tree/plugin/repository. 
I moved all spec.extension from those manifests and the test worked fine. As I will take one day off, this PR could be merged if no problems seen. 
Then CUE spec.extension could be added to enhance the `rudr addon:ls` and support install/uninstall operations. Thanks.
```
$ rudr addon:ls
NAME                            	TYPE    	DEFINITION                      	STATUS     	APPLIES-TO                                 
manualscalertrait.core.oam.dev  	trait   	manualscalertraits.core.oam.dev 	installed  	core.oam.dev/v1alpha2.ContainerizedWorkload
serviceexposes.core.oam.dev     	trait   	serviceexposes.core.oam.dev     	uninstalled	                                           
clonesetworkloads.apps.kruise.io	workload	clonesetworkloads.apps.kruise.io	uninstalled	-                                          
statefulsets.apps               	workload	statefulsets.apps               	uninstalled	-                                          

```

File: pkg/cmd/addon.go [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+		}
+		var manifestResp *http.Response
+		for _, d := range manifests {
+			var template types.Template
+			var workloadDefinition corev1alpha2.WorkloadDefinition
+			var traitDefinition corev1alpha2.TraitDefinition
+			if manifestResp, err = http.Get(d.Url); err != nil {
+				return err
+			}
+			defer manifestResp.Body.Close()
```
Comment (wonderflow) [inline pkg/cmd/addon.go:148]: we shouldn't use `defer` in a for loop

**Pull request merged**